### PR TITLE
Add ability to list preferences exposed on docs.stripe.com

### PR DESCRIPTION
**Pull request by joelzwarrington**
### Summary and motivation

Adds a `stripe docs prefs list` command that fetches and displays the user-configurable documentation preferences available on docs.stripe.com (e.g. language, theme).

CleanShot 2026-07-20 at 11 37 32

### Test plan

- [X] Changes are covered by unit tests (including failures and edge cases)
- [X] Changes were tested manually

To test manually:

```bash
make
./stripe docs prefs list
```

You should see a bullet list of available preferences, each showing the preference ID, description, allowed values, and default (if one exists).

### CODE REVIEW

File: pkg/docs/client.go [hunk 1]
```diff
@@ -178,6 +178,42 @@ func (c *Client) do(req *http.Request) (response, error) {
...
+}
+
+// PrefsResponse represents the response from the docs prefs endpoint.
+type PrefsResponse struct {
+	Prefs []Pref `json:"prefs"`
+}
+
+// FetchPrefs retrieves the list of documentation preferences from docs.stripe.com.
+func (c *Client) FetchPrefs(ctx context.Context) (*PrefsResponse, error) {
+	u := c.baseURL.JoinPath("/_endpoint/prefs")
```
Comment (matv-stripe) [inline pkg/docs/client.go:197]: would be nice to turn this into a first-party-api sometime this half instead of relying on the docs endpoint architecture, and support auth too

File: pkg/cmd/docs/prefs.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+	cmd.AddCommand(r.newPrefsListCmd())
+
+	return cmd
+}
+
+func (r *RootCommand) newPrefsListCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "list",
+		Short: "List available docs.stripe.com preferences",
```
Comment (matv-stripe) [inline pkg/cmd/docs/prefs.go]: will a user know what a "docs.stripe.com preference" means?

**Review by matv-stripe — APPROVED** [review]

File: pkg/docs/client.go (hunk 1 above)
Comment (joelzwarrington) [reply pkg/docs/client.go:197]: Yep! This is planned and being worked on. We'll be adding some proper APIs for authentication.

File: pkg/cmd/docs/prefs.go (hunk 2 above)
Comment (jil-stripe) [reply pkg/cmd/docs/prefs.go]: I have a similar question. We might want to find a more generic way of describing this.

**Review by jil-stripe — COMMENTED** [review]
Docstrings LGTM, would love someone else to review the code

**Review by rayc-stripe — APPROVED** [review]
code LGTM
